### PR TITLE
Add fontsize and font selectors

### LINKED ISSUE #1802: Selector to choose font
As mentioned here #1521, we'll have a new tab in config page, "Display". 
In this tab, we'll have a selector to choose the font size in article view.

### PR DESCRIPTION
**Pull request by nicosomb**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | yes
| CHANGELOG.md  |no
| License       | MIT

Fix #1802  

## Configuration

We can customize font family, font size, line height and maximal width. A preview text is available. 

Capture d’écran 2023-08-01 à 09 51 12

## OpenDyslexic and Atkinson Hyperlegible implementation

Capture d’écran 2023-07-29 à 13 28 10

Capture d’écran 2023-07-31 à 18 40 13

## Available fonts 

Here are the available fonts: 

```
        - 'Sans-serif'
        - 'Serif'
        - 'Atkinson Hyperlegible'
        - 'EB Garamond'
        - 'Lato'
        - 'Montserrat'
        - 'OpenDyslexicRegular'
        - 'Oswald'
```

Some of them (Atkinson Hyperlegible and OpenDyslexicRegular) are there to help people who need it to read their articles. 
Some of them (Oswald, Montserrat, Lato and EB Garamond) are embedded thanks to this study: https://www.fastcompany.com/90747518/are-fonts-ageist

For the moment, I'm not sure we have to add more fonts. 

## Move the configuration to the article view? 

As done in Omnivore or Pocket, it could be a nice improvement to move the display configuration into the article view.

⚠️  I think it's a lot of work, so we can already move forward with this pull request and create a new issue to improve it later (when we'll burn Materialize CSS). 

💡 Quick win: we can already add a new icon in the article sidebar with a link to the configuration page.

### CODE REVIEW

**Comment by nicosomb** [comment]
It could be nice to have a selector on the article view.
I can work on the server side (ajax calls, etc.) but I don't really know how we can implement that on front side. @Simounet any idea? 

See examples below: 

**Pocket**

Capture d’écran 2023-07-30 à 20 22 00

**Omnivore**

Capture d’écran 2023-07-30 à 20 22 29

**Comment by nicosomb** [comment]
## Article preview on configuration page 

Capture d’écran 2023-07-31 à 10 43 20

**Comment by nicosomb** [comment]
Assets build is fixed in this PR https://github.com/wallabag/wallabag/pull/6783#issuecomment-1658338996

Need to rebase after #6783 is merged.

**Comment by nicosomb** [comment]
Interesting review from @Kdecherf : it could be interesting to store this configuration locally, as we do for light / dark theme. 
Because this configuration is "device related" : I don't want the same display on my smartphone than on my desktop.

**Comment by romu70** [comment]
Sounds great.

I don’t get the « maximum width »settings. For a good legibility, best practices are to keep the line width from 70 to 80 chars. So, we should be able to compute this for the user once we know it’s preference for the char size, and the viewport width, correct?

**Comment by nicosomb** [comment]
> we should be able to compute this for the user once we know it’s preference for the char size

We should 😉  but I looked what our competitors have done and you can adjust the width.

**Comment by romu70** [comment]
Ok but is what the competition does a good thing?More is not always better. A good product is made of tough choices.Anyway, having a short choice of fonts like you do, is a good thing.Le 1 août 2023 à 14:17, Nicolas Lœuillet ***@***.***> a écrit :﻿

we should be able to compute this for the user once we know it’s preference for the char size

We should 😉  but I looked what our competitors have done and you can adjust the width.

File: src/Wallabag/CoreBundle/Entity/Config.php [hunk 1]
```diff
@@ -126,6 +126,51 @@ class Config
...
+     * @Groups({"config_api"})
+     */
+    private $maxWidth;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="custom_css", type="text", nullable=true)
+     *
+     * @Groups({"config_api"})
```
Comment (j0k3r) [inline src/Wallabag/CoreBundle/Entity/Config.php:170]: Should this be exposed to the API as it seems to be a calculated field? I don't think it'll be _usable_ for clients other than the web UI

File: web/wallassets/41a1d465797f51702171.otf
Comment (j0k3r) [inline web/wallassets/41a1d465797f51702171.otf:1]: What is that font? It even doesn't have a proper name

**Comment by j0k3r** [comment]
Rebase needed

File: web/wallassets/41a1d465797f51702171.otf
Comment (nicosomb) [reply web/wallassets/41a1d465797f51702171.otf:1]: It's related to `OpenDyslexic-Regular.otf`. I don't undestand the name.

**Review by j0k3r — APPROVED** [review]

**Pull request merged**